crates/ruff_server/src/session/index.rs

```diff
@@ -310,11 +309,13 @@ impl Index {
             return;
         };
 
-        // TODO: I think this does not correctly reload settings when using `extend` and the extended
-        //  setting isn't in a parent folder.
-        for (root, settings) in self.settings.range_mut(enclosing_folder.to_path_buf()..) {
+        for (root, settings) in self
+            .settings
+            .range_mut(..=enclosing_folder.to_path_buf())
+            .rev()
+        {
             if !root.starts_with(enclosing_folder) {
-                break;
+                continue;
```

Comment:
Is `break` still incorrect here? I get the impression that we could use `break` here but not entirely sure